man/systemd-stub.xml

```diff
@@ -77,7 +77,16 @@
       <listitem><para>A <literal>.splash</literal> section with an image (in the Windows
       <filename>.BMP</filename> format) to show on screen before invoking the kernel.</para></listitem>
 
-      <listitem><para>A <literal>.dtb</literal> section with a compiled binary DeviceTree.</para></listitem>
+      <listitem><para>Zero or more <literal>.dtb</literal> sections. If only one such section is present, it
+      will always be passed to the kernel. If there at least two of them, stub will first try to match the
+      DeviceTrees bundled into UKI by comparing the first <literal>compatible</literal> string in each bundled
+      DeviceTree with the first <literal>compatible</literal> string supplied by the firmware. If that fails
+      and <literal>.hwids</literal> section exists, it will be used for matching process instead. The stub
+      will use some SMBIOS values to calculate hardware IDs of the machine (that is, it will calculate hashes of
+      different combinations of these values, the same thing what <literal>fwupdtool hwids</literal> will do), then it will
+      proceed to trying to find any of them in <literal>.hwids</literal> section and will use first matching <literal>compatible</literal>
+      string as decribed for the FW-provided one. The first <literal>.dtb</literal> section with matched
```

Comment:
"first macthin compatible string" → "first matching entry's compatible string" i guess

Reply:
"string as decribed for the FW-provided one"? typo first of all, but i think i'd rephrase this as

"and will use the first matching entry's <varname>compatible</varname> string as search key among the <literal>.dtb</literal> entries, in a similar fashion as the <varname>compatible</varname> string read from the firmware provided Devicetree before".